transformers/test_data/shared_behaviors/shared_flip_bid_trigger_suite.go

```diff
@@ -0,0 +1,407 @@
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared/constants"
+	mcdStorage "github.com/makerdao/vdb-mcd-transformers/transformers/storage"
+	. "github.com/makerdao/vdb-mcd-transformers/transformers/storage/test_helpers"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/test_data"
+	"github.com/makerdao/vulcanizedb/libraries/shared/factories/event"
+	vdbStorage "github.com/makerdao/vulcanizedb/libraries/shared/factories/storage"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage/types"
+	"github.com/makerdao/vulcanizedb/pkg/core"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type BidTriggerTestInput struct {
+	Repository      vdbStorage.Repository
+	Metadata        types.ValueMetadata
+	ContractAddress string
+	TriggerTable    string
+	FieldTable      string
+	ColumnName      event.ColumnName
+	PackedValueType types.ValueType
+}
+
+func SharedBidHistoryTriggerTests(input BidTriggerTestInput) {
+	Describe(fmt.Sprintf(`updating %s trigger table`, input.TriggerTable), func() {
+		var (
+			bidID int
+			headerOne,
+			headerTwo core.Header
+			timestampOne,
+			timestampTwo string
+			diffID,
+			addressID int64
+			repo             = input.Repository
+			db               = test_config.NewTestDB(test_config.NewTestNode())
+			hashOne          = common.BytesToHash([]byte{1, 2, 3, 4, 5})
+			hashTwo          = common.BytesToHash([]byte{5, 4, 3, 2, 1})
+			getFieldQuery    = fmt.Sprintf(`SELECT "%s" FROM maker.%s ORDER BY block_number`, input.ColumnName, input.TriggerTable)
+			insertFieldQuery = fmt.Sprintf(`INSERT INTO maker.%s (address_id, block_number, bid_id, "%s", updated) VALUES ($1, $2, $3, $4, $5)`, input.TriggerTable, input.ColumnName)
+		)
+
+		BeforeEach(func() {
+			test_config.CleanTestDB(db)
+			repo.SetDB(db)
+			blockOne := rand.Int()
+			blockTwo := blockOne + 1
+			rawTimestampOne := int64(rand.Int31())
+			rawTimestampTwo := rawTimestampOne + 1
+			timestampOne = FormatTimestamp(rawTimestampOne)
+			timestampTwo = FormatTimestamp(rawTimestampTwo)
+			headerOne = CreateHeaderWithHash(hashOne.String(), rawTimestampOne, blockOne, db)
+			headerTwo = CreateHeaderWithHash(hashTwo.String(), rawTimestampTwo, blockTwo, db)
+			diffID = CreateFakeDiffRecord(db)
+			var parseErr error
+			bidID, parseErr = strconv.Atoi(input.Metadata.Keys[constants.BidId])
+			Expect(parseErr).NotTo(HaveOccurred())
+			var addressErr error
+			addressID, addressErr = shared.GetOrCreateAddress(input.ContractAddress, db)
+			Expect(addressErr).NotTo(HaveOccurred())
+		})
+
+		Describe("updating history with new value", func() {
+			It("updates field in subsequent blocks", func() {
+				_, initialColumnVal := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				_, setupErr := db.Exec(insertFieldQuery, addressID, headerTwo.BlockNumber, bidID, initialColumnVal, timestampTwo)
```

Comment:
Not sure, but do you think it would help to clarify if the test description included something like "updates field in subsequent blocks **when no associated diff records**"? That also may make things more complicated... I sometimes get too verbose with my test descriptions. 😊